components/zcash_address/src/kind/unified.rs

```diff
@@ -120,22 +120,77 @@ impl TryFrom<&[u8]> for Address {
 impl Address {
     /// Returns the raw encoding of this Unified Address.
     pub(crate) fn to_bytes(&self) -> Vec<u8> {
-        self.0
+        let encoded: Vec<_> = self
+            .0
             .iter()
             .flat_map(|receiver| {
                 let addr = receiver.addr();
                 // Holds by construction.
                 assert!(addr.len() < 256);
 
-                let encoded: Vec<_> = iter::empty()
+                iter::empty()
                     .chain(Some(receiver.typecode()))
                     .chain(Some(addr.len() as u8))
                     .chain(addr.into_iter().cloned())
-                    .chain(iter::repeat(0).take(PADDING_LEN))
-                    .collect();
-
-                f4jumble::f4jumble(&encoded).unwrap()
             })
-            .collect()
+            .chain(iter::repeat(0).take(PADDING_LEN))
+            .collect();
+
+        f4jumble::f4jumble(&encoded).unwrap()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::convert::TryFrom;
+
+    use proptest::{
+        array::{uniform11, uniform20, uniform32},
+        prelude::*,
+    };
+
+    use super::{Address, Receiver};
+
+    prop_compose! {
+        fn uniform43()(a in uniform11(0u8..), b in uniform32(0u8..)) -> [u8; 43] {
```

Comment:
Oh, this is a nice technique for larger byte arrays, I should have thought of this before! I've used `vec(any::<u8>(), size)` before but then had to convert.

Reply:
One downside to this is the separate halves shrink separately (though that's less of an issue here given that these addresses _are_ composite types). In theory we can implement our own custom array sizes directly, but I couldn't figure out how to set up the `UniformArrayStrategy` type properly.